packages/backend/src/apps/lettersg/actions/create-letter/index.ts

```diff
@@ -0,0 +1,123 @@
+import type { IRawAction } from '@plumber/types'
+
+import { ZodError } from 'zod'
+import { fromZodError } from 'zod-validation-error'
+
+import StepError, { GenericSolution } from '@/errors/step'
+
+import { requestSchema, responseSchema } from './schema'
+
+// TBC
+type LettersApiErrorData = {
+  success: boolean
+  message: string
+  errors: Record<string, string>[]
+}
+
+const action: IRawAction = {
+  name: 'Create Letter',
+  key: 'createLetter',
+  description: 'Create a new letter based on the template id input',
+  arguments: [
+    {
+      label: 'Template Id',
+      key: 'templateId',
+      placeholder: 'Template',
+      type: 'dropdown' as const,
+      required: true,
+      description: 'Choose the template id you want for creating a letter.',
+      variables: false,
+      source: {
+        type: 'query' as const,
+        name: 'getDynamicData' as const,
+        arguments: [
+          {
+            name: 'key',
+            value: 'getTemplateIds',
+          },
+        ],
+      },
+    },
+    {
+      label: 'Letter Parameters',
+      key: 'letterParams',
+      type: 'multirow' as const,
+      required: false,
+      description:
+        'Specify every field name and values for your letter template.',
```

Comment:
might this wording be slightly better?
```suggestion
        'Specify each field name and value for your letter template.',
```